app/models/concerns/job_management.rb

```diff
@@ -15,7 +15,9 @@ def start_job
   end
 
   def destroy_job
-    destroy if dray_client.delete_job(self.key)
+    dray_client.delete_job(self.key)
+  ensure
+    destroy
```

Comment:
Instead of having `:destroy_job` as a separate method from `:destroy` should we just wire up something to the `after_destroy` callback that invokes `dray_client.delete_job`? I think we've used that pattern in a couple other places already.